front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx

```diff
@@ -185,6 +187,13 @@ const PowerRestrictionsSelector = ({
     }
   }, [powerRestrictionRanges]);
 
+  const formattedIntervals = useMemo(
+    () => formatIntervals(formattedPathElectrificationRanges, powerRestrictionRanges),
+    [formattedPathElectrificationRanges, powerRestrictionRanges]
+  );
+
+  const [useFormattedIntervals, setUseFormattedIntervals] = useState(true);
```

Comment:
You can replace all this code by a useEffect, it's simpler :+1: 
```suggestion
   useEffect(() => {
    const formattedPowerRestrictionRanges = formatIntervals(
      formattedPathElectrificationRanges,
      powerRestrictionRanges
    );
    console.log({ powerRestrictionRanges, formattedPowerRestrictionRanges });
    dispatch(updatePowerRestrictionRanges(formattedPowerRestrictionRanges));
  }, [formattedPathElectrificationRanges]);
```

Reply:
fixed